chainservice/builder.go

```diff
@@ -645,13 +649,27 @@ func (builder *Builder) registerRollDPoSProtocol() error {
 			return blockchain.Productivity(chain, start, end)
 		},
 		dao.GetBlockHash,
+		getBlockTime,
 	)
 	if err != nil {
 		return errors.Wrap(err, "failed to generate poll protocol")
 	}
 	return pollProtocol.Register(builder.cs.registry)
 }
 
+func (builder *Builder) buildGetBlockTime() (err error) {
```

Comment:
`buildBlockTimeCalculator` is a better name

Reply:
agree